### PR TITLE
Add a table of content to the config file wiki

### LINKED ISSUE #4451: Improvement over the config task wiki page
This is not a request for the bot, but for the wiki.  More precisely, the config file wiki page: https://github.com/PokemonGoF/PokemonGo-Bot/blob/dev/docs/configuration_files.md

I think this page could benefit from improvements (both from the user and developer perspective)

### Possible solution

- From a user perspective, in order to find an information about a task, you either (1) scroll over 500 lines or (2) use the search tool from your browser. This is an issue, since those workaround do not directly direct the user toward the information. A table of content may solve this issue.

- From a developer perspective, all the configuration of all the tasks are gathered in one file. This is an issue for developers, since it doesn't allow to keep track and clearly separate the information. A inclusion of files (one task = one file) into another file may solve this issue.

I know the default md format of github doesn't allow for an automatic table of content or even an include tag, but I'm pretty sure others options exists. :)

### How it would help others

A healthy wiki is a healthy community (I just made that up, but you good the idea) :)

### PR DESCRIPTION
**Pull request by supercourgette**
## Short Description:

Add a table of content to the configuration file task. 

It is not beautiful, it is not working entirely (because of md, when an header has the same name than another header, the anchor links always go to the first one), but it is something. 

The table was generated with markdown-toclify, which I added to the pip config (https://github.com/rasbt/markdown-toclify). I don't know if you feel like keeping it or no...

## Fixes/Resolves/Closes (please use correct syntax):
- Closes #4451 (partially)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -24,3 +24,4 @@ raven==5.23.0
 demjson==2.2.4
 greenlet==0.4.9
 yoyo-migrations==5.0.3
+markdown_toclify==0.1.7
```
Comment (nivong) [inline requirements.txt]: why is this necessary ?

File: docs/configuration_files.md [hunk 2]
```diff
@@ -1,9 +1,65 @@
-## Usage (up-to-date)
-  1. copy `config.json.example` to `config.json`.
-  2. Edit `config.json` and replace `auth_service`, `username`, `password`, `location` and `gmapkey` with your parameters (other keys are optional, check `Advance Configuration` below)
-  3. Simply launch the script with : `./run.sh` or `./pokecli.py` or `python pokecli.py -cf ./configs/config.json` if you want to specify a config file
-
+<a class="mk-toclify" id="table-of-contents"></a>
```
Comment (TheSavior) [inline docs/configuration_files.md]: I'm not sure why we want these `a` tags.

File: docs/configuration_files.md [hunk 3]
```diff
@@ -24,111 +80,123 @@
...
-    },
-    {
-      "type": "MoveToFort"
-    },
-    {
-      "type": "FollowSpiral"
-    }
-  ]
-  // ...
+// ...
```
Comment (TheSavior) [inline docs/configuration_files.md]: The whitespace got killed in these examples.

**Comment by nivong** [comment]
👎 please double check your code a lot of html tags and a lot of things that needs optimizing

File: requirements.txt (hunk 1 above)
Comment (TheSavior) [inline requirements.txt]: It seems unlikely that we would want to add a dependency like this.

Comment (supercourgette) [inline requirements.txt]: A table of content allows for an effective way to find the needed information. I ll admit this one is at the early stage...

**Comment by supercourgette** [comment]
I'll double check everything and will have look at the <a>. However being able to go back to the top is useful on such a huge page.

File: requirements.txt (hunk 1 above)
Comment (supercourgette) [inline requirements.txt]: I will remove it, sorry. :)

**Comment by supercourgette** [comment]
I deeply apologize for the breaking of the config files. Everything should be back to normal now. I was tired I guess, but I still can't believe I didn't even notice the obvious missing space, the arrays, ...

**Comment by mjmadsen** [comment]
Thanks for the PR
